src/openrct2/game.c

```diff
@@ -1072,7 +1072,12 @@ void game_fix_save_vars()
     for (sint32 i = 0; i < MAX_RESEARCH_ITEMS; i++) {
         rct_research_item *researchItem = &gResearchItems[i];
         if (researchItem->entryIndex == RESEARCHED_ITEMS_SEPARATOR) continue;
-        if (researchItem->entryIndex == RESEARCHED_ITEMS_END) continue;
+        if (researchItem->entryIndex == RESEARCHED_ITEMS_END)
+        {
+            assert(i < (MAX_RESEARCH_ITEMS - 1));
```

Comment:
Don't assert, the idea of this function is to fix invalid save files - so it should assume this can happen and do an appropriate fix.